erpnext/accounts/report/profitability_analysis/profitability_analysis.py

```diff
@@ -0,0 +1,188 @@
+		return None
+
+	accounts, accounts_by_name, parent_children_map = filter_accounts(accounts)
+
+	gl_entries_by_account = {}
+
+	set_gl_entries_by_account(filters.company, filters.from_date,
+		filters.to_date, based_on, gl_entries_by_account, ignore_closing_entries=not flt(filters.with_period_closing_entry))
+
+	total_row = calculate_values(accounts, gl_entries_by_account, filters)
+	accumulate_values_into_parents(accounts, accounts_by_name)
+
+	data = prepare_data(accounts, filters, total_row, parent_children_map, based_on)
+	data = filter_out_zero_value_rows(data, parent_children_map, 
+		show_zero_values=filters.get("show_zero_values"))
+
+	return data
+
+def calculate_values(accounts, gl_entries_by_account, filters):
+	init = {
+		"income": 0.0,
+		"expense": 0.0,
+		"total": 0.0
+	}
+
+	total_row = {
+		"cost_center": None,
+		"account_name": "'" + _("Total") + "'",
+		"warn_if_negative": True,
+		"income": 0.0,
+		"expense": 0.0,
+		"total": 0.0
+	}
+
+	for d in accounts:
+		d.update(init.copy())
+
+		# add opening
+
+		for entry in gl_entries_by_account.get(d.name, []):
+			if cstr(entry.is_opening) != "Yes":
+				if entry.type == 'Income':
+					d["income"] += flt(entry.credit) - flt(entry.debit)
+				if entry.type == 'Expense':
+					d["expense"] += flt(entry.debit) - flt(entry.credit)
+
+				d["total"] = d.get("income") - d.get("expense")
+
+		total_row["income"] += d["income"]
+		total_row["expense"] += d["expense"]
+
+	total_row["total"] = total_row.get("income") - total_row.get("expense")
+
+	return total_row
+
+def accumulate_values_into_parents(accounts, accounts_by_name):
+	for d in reversed(accounts):
+		if d.parent_account:
+			for key in value_fields:
+				accounts_by_name[d.parent_account][key] += d[key]
+
+def prepare_data(accounts, filters, total_row, parent_children_map, based_on):
+	data = []
+	company_currency = frappe.db.get_value("Company", filters.company, "default_currency")
+
+	for d in accounts:
+		has_value = False
+		row = {
+			"account_name": d.account_name or d.name,
+			"account": d.name,
+			"parent_account": d.parent_account,
+			"indent": d.indent,
+			"fiscal_year": filters.fiscal_year,
+			"currency": company_currency,
+			"based_on": based_on
+		}
+
+		for key in value_fields:
+			row[key] = flt(d.get(key, 0.0), 3)
+			
+			if abs(row[key]) >= 0.005:
+				# ignore zero values
+				has_value = True
+
+		row["has_value"] = has_value
+		data.append(row)
+		
+	data.extend([{},total_row])
+
+	return data
+
+def get_columns(filters):
+	return [
+		{
+			"fieldname": "account",
+			"label": _(filters.based_on),
+			"fieldtype": "Link",
+			"options": filters.based_on,
+			"width": 300
+		},
+		{
+			"fieldname": "income",
+			"label": _("Income"),
+			"fieldtype": "Currency",
+			"options": "currency",
+			"width": 120
+		},
+		{
+			"fieldname": "expense",
+			"label": _("Expense"),
+			"fieldtype": "Currency",
+			"options": "currency",
+			"width": 120
+		},
+		{
+			"fieldname": "total",
```

Comment:
Use fieldname as per label "gross_profit_loss", it will help a lot to debug in the future.